unit/getinfo_test.c

```diff
@@ -178,51 +457,74 @@ fail:										\
 getinfo_test(src, 3, "Test with service, FI_SOURCE flag and no node",
 		 NULL, opts.src_port, FI_SOURCE, hints, NULL,
-		 check_srcaddr, 0)	// TODO should we check for wildcard addr?
+		 NULL, check_srcaddr, 0)	// TODO should we check for wildcard addr?
 getinfo_test(src, 4, "Test with node, service, FI_SOURCE flags",
 		opts.src_addr ? opts.src_addr : "localhost", opts.src_port,
-		FI_SOURCE, hints, NULL, check_srcaddr, 0)
+		FI_SOURCE, hints, NULL, NULL, check_srcaddr, 0)
 
 /* 2.2 Source and destination address tests */
 getinfo_test(src_dest, 1, "Test with node, service",
 		opts.dst_addr ? opts.dst_addr : "localhost", opts.dst_port,
-		0, hints, NULL, check_src_dest_addr, 0)
+		0, hints, NULL, NULL, check_src_dest_addr, 0)
 
 getinfo_test(src_dest, 2, "Test API version",
-		NULL, NULL, 0, hints, NULL, check_api_version ,0)
+		NULL, NULL, 0, hints, NULL, NULL, check_api_version , 0)
 
 /* Negative tests */
 getinfo_test(neg, 1, "Test with non-existent domain name",
-		NULL, NULL, 0, hints, invalid_dom, NULL, -FI_ENODATA)
+		NULL, NULL, 0, hints, invalid_dom, NULL, NULL, -FI_ENODATA)
 
 /* Utility provider tests */
 getinfo_test(util, 1, "Test if we get utility provider when requested",
-		NULL, NULL, 0, hints, NULL, check_util_prov, 0)
+		NULL, NULL, 0, hints, NULL, NULL, check_util_prov, 0)
+
+/* Message Ordering Tests */
+getinfo_test(ordering, 1, "Test msg ordering bits supported are set",
+		NULL, NULL, 0, hints, NULL, validate_msg_ordering_bits, NULL, 0)
+getinfo_test(ordering, 2, "Test rma RAW ordering size is set",
+		NULL, NULL, 0, hints, init_valid_rma_RAW_ordering_no_set_size, NULL, check_valid_rma_ordering_sizes, 0)
+getinfo_test(ordering, 3, "Test rma RAW ordering size is set to hints",
+		NULL, NULL, 0, hints, init_valid_rma_RAW_ordering_set_size, NULL, check_valid_rma_ordering_sizes, 0)
+getinfo_test(ordering, 4, "Test rma WAR ordering size is set",
+		NULL, NULL, 0, hints, init_valid_rma_WAR_ordering_no_set_size, NULL, check_valid_rma_ordering_sizes, 0)
+getinfo_test(ordering, 5, "Test rma WAR ordering size is set to hints",
+		NULL, NULL, 0, hints, init_valid_rma_WAR_ordering_set_size, NULL, check_valid_rma_ordering_sizes, 0)
+getinfo_test(ordering, 6, "Test rma WAW ordering size is set",
+		NULL, NULL, 0, hints, init_valid_rma_WAW_ordering_no_set_size, NULL, check_valid_rma_ordering_sizes, 0)
+getinfo_test(ordering, 7, "Test rma WAW ordering size is set to hints",
+		NULL, NULL, 0, hints, init_valid_rma_WAW_ordering_set_size, NULL, check_valid_rma_ordering_sizes, 0)
+getinfo_test(ordering, 8, "Test invalid rma RAW ordering size",
+		NULL, NULL, 0, hints, init_invalid_rma_RAW_ordering_size, NULL, NULL, -FI_ENODATA)
+getinfo_test(ordering, 9, "Test invalid rma WAR ordering size",
+		NULL, NULL, 0, hints, init_invalid_rma_WAR_ordering_size, NULL, NULL, -FI_ENODATA)
+getinfo_test(ordering, 10, "Test invalid rma WAW ordering size",
+		NULL, NULL, 0, hints, init_invalid_rma_WAW_ordering_size, NULL, NULL, -FI_ENODATA)
```

Comment:
@nrspruit will be out, so this is mostly for my reference in merging this.  The line lengths are too long, and we need better names than ordering1, ordering2, etc.  I will apply those cleanups either after merging or merge manually.